Header: Make search icon consistent between desktop and mobile

iandunn opened the pull request:
Fixes #369

This keeps the desktop styles, and makes the mobile icon smaller and center-positioned. I think the size is correct, but I'm not sure what the precise alignment should be. Let me know if you'd like changes.

<img width="648" alt="Screenshot 2023-03-28 at 4 22 10 PM" src="https://user-images.githubusercontent.com/484068/228388515-8ebbe7d9-a7e1-4647-9ad2-c93e820cb5c8.png">

iandunn committed:
Header: Make search icon consistent between desktop and mobile

This keeps the desktop styles, and makes the mobile icon smaller and center-positioned.

jasmussen commented:
Thanks for the PR, it certainly looks better. I delved into the inspector more, and it appears that the SVG is used as a background image, and comes with a somewhat nonstandard size (20x19) compared to the [new WordPress icon library](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library) (24x24). The menu icon is correct, though. 

Here's a comparison. The two right most are from the library:

<img width="154" alt="Frame 3" src="https://user-images.githubusercontent.com/1204802/228454502-404fe5ae-5129-4dba-bec8-29ac31019bec.png">

Here it is, zoomed in:

<img width="781" alt="Screenshot 2023-03-29 at 09 10 58" src="https://user-images.githubusercontent.com/1204802/228454537-17f8182c-1c85-4807-9bff-5198832c5550.png">

So this needs more work on my end:

* The core icon is the one we should use, it has the correct 24x24 footprint and uses the right stroke width.
* However it's not optically centered horizontally.

jasmussen commented:
Upstream fix: https://github.com/WordPress/gutenberg/pull/49425

jasmussen commented:
There's a little discussion in the upstream fix, ~but I think it might result in this SVG~ it did result in this SVG:

```
<svg xmlns="http://www.w3.org/2000/svg" xml:space="preserve" style="enable-background:new 0 0 24 24" viewBox="0 0 24 24"><path d="M13 5c-3.3 0-6 2.7-6 6 0 1.4.5 2.7 1.3 3.7l-3.8 3.8 1.1 1.1 3.8-3.8c1 .8 2.3 1.3 3.7 1.3 3.3 0 6-2.7 6-6S16.3 5 13 5zm0 10.5c-2.5 0-4.5-2-4.5-4.5s2-4.5 4.5-4.5 4.5 2 4.5 4.5-2 4.5-4.5 4.5z" style="fill-rule:evenodd;clip-rule:evenodd;fill:#1e1e1e"/></svg>
```

iandunn committed:
Update SVG to match Core

can't use core directly b/c has to be bg image b/c using nav block, and bg images can use fill/stroke

iandunn commented:
Thanks for that, I've switched out the SVG and it's using the `24x24` footprint now:

![Screenshot 2023-03-29 at 3 48 47 PM](https://user-images.githubusercontent.com/484068/228685104-8f56c52d-6eaf-4967-bd87-dd7070c0675b.png)

![Screenshot 2023-03-29 at 3 48 40 PM](https://user-images.githubusercontent.com/484068/228685102-d599d3ab-f0a4-4d71-8646-7908d6a86470.png)

The reason it's a background image is because we're using the Search block inside the Navigation block:

https://github.com/WordPress/wporg-mu-plugins/blob/b32eb9e1fe74396c0d2fca60c58f68e3ada247de/mu-plugins/blocks/global-header-footer/header.php#L96-L103

AFAIK we can't change the markup of the Search block to swap in a different image, so we used `background-image` instead. One downside is that we need to fork the Core image in order to set the `fill`, but that seems much simpler to me than the alternative. Do you see a compelling reason to change that?

jasmussen commented:
Looks good, go for it!

iandunn commented:
👍🏻
